config-model-api/src/main/java/com/yahoo/config/model/api/ConfigChangeReindexAction.java

```diff
@@ -0,0 +1,20 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.config.model.api;
+
+import java.util.Optional;
+
+/**
+ * Represents an action to re-index a document type in order to handle a config change.
+ *
+ * @author bjorncs
+ */
+public interface ConfigChangeReindexAction extends ConfigChangeAction {
+
+    @Override default Type getType() { return Type.REINDEXING; }
+
+    /** @return name identifying this kind of change, used to identify names which should be allowed */
+    String name();
+
+    /** @return name of the document type that must bed re-indexed, or empty if all document types */
+    Optional<String> getDocumentType();
```

Comment:
Hmm, is "all document types" an option here?

Reply:
I'm not sure, most likely not. The programmatic API allows it to be null. It is first null and then later updated with document type (uncertain whether this always happen).